File: kt/task/01begin.kt
package task

import java.lang.Math.pow
import kotlin.math.abs
import kotlin.math.sqrt

class BeginTask {

    var datScanner = scanner.Scanner()
    var ansScanner = scanner.Scanner()

    fun solve(pathPrefix: String) {
        for (taskNo in 1..40) {
            var taskNoStr = "%03d".format(taskNo)
            var fileName: String
            for (testNo in 1..100) {
                var testNoStr = "%03d".format(testNo)
                fileName = "$pathPrefix/01begin/Begin$taskNoStr/$testNoStr/$testNoStr"
                if (!datScanner.changeReader("$fileName.dat")) {
                    println("Begin$taskNoStr couldn't open datScanner source")
                    return
                }
                if (!ansScanner.changeReader("$fileName.ans")) {
                    println("Begin$taskNoStr couldn't open ansScanner source")
                    return
                }
                val ok = when (taskNo) {
                    1 -> this.begin1()
                    2 -> this.begin2()
                    3 -> this.begin3()
                    4 -> this.begin4()
                    5 -> this.begin5()
                    6 -> this.begin6()
                    7 -> this.begin7()
                    8 -> this.begin8()
                    9 -> this.begin9()
                    10 -> this.begin10()
                    11 -> this.begin11()
                    12 -> this.begin12()
                    13 -> this.begin13()
                    14 -> this.begin14()
                    15 -> this.begin15()
                    16 -> this.begin16()
                    17 -> this.begin17()
                    18 -> this.begin18()
                    19 -> this.begin19()
                    20 -> this.begin20()
                    21 -> this.begin21()
                    22 -> this.begin22()
                    23 -> this.begin23()
                    24 -> this.begin24()
                    25 -> this.begin25()
                    26 -> this.begin26()
                    27 -> this.begin27()
                    28 -> this.begin28()
                    29 -> this.begin29()
                    30 -> this.begin30()
                    31 -> this.begin31()
                    32 -> this.begin32()
                    33 -> this.begin33()
                    34 -> this.begin34()
                    35 -> this.begin35()
                    36 -> this.begin36()
                    37 -> this.begin37()
                    38 -> this.begin38()
                    39 -> this.begin39()
                    40 -> this.begin40()
                    else -> false
                }
                scanner.removeFiles()
                if (!ok) {
                    println("Begin$taskNoStr Test#$testNoStr has not been tested")
                    return
                }
                if (!datScanner.eof()) {
                    println("Begin$taskNoStr Test#$testNoStr input is not enough")
                    return
                }
                if (!ansScanner.eof()) {
                    println("Begin$taskNoStr Test#$testNoStr output is not enough")
                    return
                }
            }
            println("Begin$taskNoStr has been tested!")
        }
    }

    private fun begin1(): Boolean {
        val a = this.datScanner.nextDouble()
        val p = 4 * a
        return this.ansScanner.checkDouble(2, p)
    }

    private fun begin2(): Boolean {
        val a = this.datScanner.nextDouble()
        val s = a * a
        return this.ansScanner.checkDouble(2, s)
    }

    private fun begin3(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val s = a * b
        val p = 2 * (a + b)
        return this.ansScanner.checkDouble(2, s, p)
    }

    private fun begin4(): Boolean {
        val d = this.datScanner.nextDouble()
        val pi = 3.14
        val l = pi * d
        return this.ansScanner.checkDouble(2, l)
    }

    private fun begin5(): Boolean {
        val a = this.datScanner.nextDouble()
        val v = a * a * a
        val s = 6 * a * a
        return this.ansScanner.checkDouble(3, v, s)
    }

    private fun begin6(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = this.datScanner.nextDouble()
        val v = a * b * c
        val s = 2 * (a * b + b * c + c * a)
        return this.ansScanner.checkDouble(3, v, s)
    }

    private fun begin7(): Boolean {
        val r = this.datScanner.nextDouble()
        val pi = 3.14
        val l = 2 * pi * r
        val s = pi * r * r
        return this.ansScanner.checkDouble(3, l, s)
    }

    private fun begin8(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val average = (a + b) / 2
        return this.ansScanner.checkDouble(2, average)
    }

    private fun begin9(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val gMean = sqrt(a * b)
        return this.ansScanner.checkDouble(2, gMean)
    }

    private fun begin10(): Boolean {
        var a = this.datScanner.nextDouble()
        var b = this.datScanner.nextDouble()
        a *= a
        b *= b
        val sum = a + b
        val sub = a - b
        val mul = a * b
        val div = a / b
        return this.ansScanner.checkDouble(2, sum, sub, mul, div)
    }

    private fun begin11(): Boolean {
        var a = this.datScanner.nextDouble()
        var b = this.datScanner.nextDouble()
        a = abs(a)
        b = abs(b)
        val sum = a + b
        val sub = a - b
        val mul = a * b
        val div = a / b
        return this.ansScanner.checkDouble(2, sum, sub, mul, div)
    }

    private fun begin12(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = sqrt(a * a + b * b)
        val p = a + b + c
        return this.ansScanner.checkDouble(2, c, p)
    }

    private fun begin13(): Boolean {
        val r1 = this.datScanner.nextDouble()
        val r2 = this.datScanner.nextDouble()
        val pi = 3.14
        val s1 = pi * r1 * r1
        val s2 = pi * r2 * r2
        val s3 = abs(s1 - s2)
        return this.ansScanner.checkDouble(2, s1, s2, s3)
    }

    private fun begin14(): Boolean {
        val l = this.datScanner.nextDouble()
        val pi = 3.14
        val r = l / (2 * pi)
        val s = pi * r * r
        return this.ansScanner.checkDouble(2, r, s)
    }

    private fun begin15(): Boolean {
        val s = this.datScanner.nextDouble()
        val pi = 3.14
        val d = sqrt(4 * s / pi)
        val l = pi * d
        return this.ansScanner.checkDouble(2, d, l)
    }

    private fun begin16(): Boolean {
        val x1 = this.datScanner.nextDouble()
        val x2 = this.datScanner.nextDouble()
        val distance = abs(x2 - x1)
        return this.ansScanner.checkDouble(2, distance)
    }

    private fun begin17(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = this.datScanner.nextDouble()
        val ac = abs(c - a)
        val bc = abs(c - b)
        val sum = ac + bc
        return this.ansScanner.checkDouble(2, ac, bc, sum)
    }

    private fun begin18(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = this.datScanner.nextDouble()
        val ac = abs(c - a)
        val bc = abs(c - b)
        val mul = ac * bc
        return this.ansScanner.checkDouble(2, mul)
    }

    private fun begin19(): Boolean {
        val x1 = this.datScanner.nextDouble()
        val y1 = this.datScanner.nextDouble()
        val x2 = this.datScanner.nextDouble()
        val y2 = this.datScanner.nextDouble()
        val a = abs(x2 - x1)
        val b = abs(y2 - y1)
        val p = 2 * (a + b)
        val s = a * b
        return this.ansScanner.checkDouble(2, p, s)
    }

    private fun begin20(): Boolean {
        val x1 = this.datScanner.nextDouble()
        val y1 = this.datScanner.nextDouble()
        val x2 = this.datScanner.nextDouble()
        val y2 = this.datScanner.nextDouble()
        val distance = sqrt(pow(x2 - x1, 2.0) + pow(y2 - y1, 2.0))
        return this.ansScanner.checkDouble(2, distance)
    }

    private fun begin21(): Boolean {
        val x1 = this.datScanner.nextDouble()
        val y1 = this.datScanner.nextDouble()
        val x2 = this.datScanner.nextDouble()
        val y2 = this.datScanner.nextDouble()
        val x3 = this.datScanner.nextDouble()
        val y3 = this.datScanner.nextDouble()
        val a = sqrt(pow(x2 - x1, 2.0) + pow(y2 - y1, 2.0))
        val b = sqrt(pow(x3 - x2, 2.0) + pow(y3 - y2, 2.0))
        val c = sqrt(pow(x3 - x1, 2.0) + pow(y3 - y1, 2.0))
        val p = a + b + c
        val halfP = p / 2
        val s = sqrt(halfP * (halfP - a) * (halfP - b) * (halfP - c))
        return this.ansScanner.checkDouble(2, p, s)
    }

    private fun begin22(): Boolean {
        var a = this.datScanner.nextDouble()
        var b = this.datScanner.nextDouble()
        a += b
        b = a - b
        a -= b
        return this.ansScanner.checkDouble(2, a, b)
    }

    private fun begin23(): Boolean {
        var a = this.datScanner.nextDouble()
        var b = this.datScanner.nextDouble()
        var c = this.datScanner.nextDouble()
        val d = a
        a = c
        c = b
        b = d
        return this.ansScanner.checkDouble(2, a, b, c)
    }

    private fun begin24(): Boolean {
        var a = this.datScanner.nextDouble()
        var b = this.datScanner.nextDouble()
        var c = this.datScanner.nextDouble()
        val d = a
        a = b
        b = c
        c = d
        return this.ansScanner.checkDouble(2, a, b, c)
    }

    private fun begin25(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = 3 * pow(x, 6.0) - 6 * x * x - 7
        return this.ansScanner.checkDouble(2, y)
    }

    private fun begin26(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = 4 * pow(x - 3, 6.0) - 7 * pow(x - 3, 3.0) + 2
        return this.ansScanner.checkDouble(2, y)
    }

    private fun begin27(): Boolean {
        val a = this.datScanner.nextDouble()
        val degA2 = a * a
        val degA4 = degA2 * degA2
        val degA8 = degA4 * degA4
        return this.ansScanner.checkDouble(2, degA2, degA4, degA8)
    }

    private fun begin28(): Boolean {
        val a = this.datScanner.nextDouble()
        val degA2 = a * a
        val degA3 = degA2 * a
        val degA5 = degA3 * degA2
        val degA10 = degA5 * degA5
        val degA15 = degA10 * degA5
        return this.ansScanner.checkDouble(2, degA2, degA3, degA5, degA10, degA15)
    }

    private fun begin29(): Boolean {
        val degrees = this.datScanner.nextDouble()
        val pi = 3.14
        val radians = degrees * pi / 180
        return this.ansScanner.checkDouble(2, radians)
    }

    private fun begin30(): Boolean {
        val radians = this.datScanner.nextDouble()
        val pi = 3.14
        val degrees = 180 * radians / pi
        return this.ansScanner.checkDouble(2, degrees)
    }

    private fun begin31(): Boolean {
        val tFahrenheit = this.datScanner.nextDouble()
        val tCentigrade = (tFahrenheit - 32) / 9 * 5
        return this.ansScanner.checkDouble(2, tCentigrade)
    }

    private fun begin32(): Boolean {
        val tCentigrade = this.datScanner.nextDouble()
        val tFahrenheit = tCentigrade / 5 * 9 + 32
        return this.ansScanner.checkDouble(2, tFahrenheit)
    }

    private fun begin33(): Boolean {
        val x = this.datScanner.nextDouble()
        val a = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val oneKg = a / x
        val yKgPrice = oneKg * y
        return this.ansScanner.checkDouble(2, oneKg, yKgPrice)
    }

    private fun begin34(): Boolean {
        val x = this.datScanner.nextDouble()
        val a = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val oneKgChocoPrice = a / x
        val oneKgCandyPrice = b / y
        val expensive = oneKgChocoPrice / oneKgCandyPrice
        return this.ansScanner.checkDouble(2, oneKgChocoPrice, oneKgCandyPrice, expensive)
    }

    private fun begin35(): Boolean {
        val v = this.datScanner.nextDouble()
        val u = this.datScanner.nextDouble()
        val t1 = this.datScanner.nextDouble()
        val t2 = this.datScanner.nextDouble()
        val s = v * t1 + (v - u) * t2
        return this.ansScanner.checkDouble(2, s)
    }

    private fun begin36(): Boolean {
        val v1 = this.datScanner.nextDouble()
        val v2 = this.datScanner.nextDouble()
        var s = this.datScanner.nextDouble()
        val t = this.datScanner.nextDouble()
        s += t * (v1 + v2)
        return this.ansScanner.checkDouble(2, s)
    }

    private fun begin37(): Boolean {
        val v1 = this.datScanner.nextDouble()
        val v2 = this.datScanner.nextDouble()
        var s = this.datScanner.nextDouble()
        val t = this.datScanner.nextDouble()
        s = abs(s - t * (v1 + v2))
        return this.ansScanner.checkDouble(2, s)
    }

    private fun begin38(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val x = -b / a
        return this.ansScanner.checkDouble(2, x)
    }

    private fun begin39(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = this.datScanner.nextDouble()
        val d = b * b - 4 * a * c
        val x1 = (-b - sqrt(d)) / 2 / a
        val x2 = (-b + sqrt(d)) / 2 / a
        return this.ansScanner.checkDouble(2, x1, x2)
    }

    private fun begin40(): Boolean {
        val a1 = this.datScanner.nextDouble()
        val b1 = this.datScanner.nextDouble()
        val c1 = this.datScanner.nextDouble()
        val a2 = this.datScanner.nextDouble()
        val b2 = this.datScanner.nextDouble()
        val c2 = this.datScanner.nextDouble()
        val d = a1 * b2 - a2 * b1
        val x = (c1 * b2 - c2 * b1) / d
        val y = (a1 * c2 - a2 * c1) / d
        return this.ansScanner.checkDouble(2, x, y)
    }
}

File: go/task/01begin.go
package task

import (
	"fmt"
	"math"

	"github.com/pkg/errors"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/checker"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/scanner"
)

type beginTask struct {
	dataPath string
	name     string
	count    int
	scanner  scanner.Scanner
	checker  checker.Checker
}

func NewBeginTask(pathPrefix string) Task {
	return &beginTask{
		dataPath: pathPrefix + "/01begin/Begin%03d/%03d/%03d",
		name:     "Begin",
		count:    40,
	}
}

func (bt *beginTask) GetCount() int { return bt.count }

func (bt *beginTask) GetName() string { return bt.name }

func (bt *beginTask) SetData(taskNo int, testNo int) error {
	var filePath = fmt.Sprintf(bt.dataPath, taskNo, testNo, testNo)
	scanner, err := scanner.New(filePath + ".dat")
	if err != nil {
		return errors.Wrap(err, "scanner.New")
	}
	checker, err := checker.New(filePath + ".ans")
	if err != nil {
		return errors.Wrap(err, "checker.New")
	}
	bt.scanner = scanner
	bt.checker = checker
	return nil
}

func (bt *beginTask) ScannerEOF() bool { return bt.scanner.EOF() }

func (bt *beginTask) CheckerEOF() bool { return bt.checker.EOF() }

func (bt *beginTask) CleanData() {
	bt.scanner.RemoveFiles()
	bt.checker.RemoveFiles()
}

func (bt *beginTask) GetFn(taskNo int) func() bool {
	switch taskNo {
	case 1:
		return bt.begin1
	case 2:
		return bt.begin2
	case 3:
		return bt.begin3
	case 4:
		return bt.begin4
	case 5:
		return bt.begin5
	case 6:
		return bt.begin6
	case 7:
		return bt.begin7
	case 8:
		return bt.begin8
	case 9:
		return bt.begin9
	case 10:
		return bt.begin10
	case 11:
		return bt.begin11
	case 12:
		return bt.begin12
	case 13:
		return bt.begin13
	case 14:
		return bt.begin14
	case 15:
		return bt.begin15
	case 16:
		return bt.begin16
	case 17:
		return bt.begin17
	case 18:
		return bt.begin18
	case 19:
		return bt.begin19
	case 20:
		return bt.begin20
	case 21:
		return bt.begin21
	case 22:
		return bt.begin22
	case 23:
		return bt.begin23
	case 24:
		return bt.begin24
	case 25:
		return bt.begin25
	case 26:
		return bt.begin26
	case 27:
		return bt.begin27
	case 28:
		return bt.begin28
	case 29:
		return bt.begin29
	case 30:
		return bt.begin30
	case 31:
		return bt.begin31
	case 32:
		return bt.begin32
	case 33:
		return bt.begin33
	case 34:
		return bt.begin34
	case 35:
		return bt.begin35
	case 36:
		return bt.begin36
	case 37:
		return bt.begin37
	case 38:
		return bt.begin38
	case 39:
		return bt.begin39
	case 40:
		return bt.begin40
	default:
		return nil
	}
}

// { "no": 1, "dat": "2", "ans": "2" }
func (bt *beginTask) begin1() bool {
	var a = bt.scanner.NextFloat64()
	var p = 4 * a
	return bt.checker.CompareFloat64(2, p)
}

// { "no": 2, "dat": "2", "ans": "2" }
func (bt *beginTask) begin2() bool {
	var a = bt.scanner.NextFloat64()
	var s = a * a
	return bt.checker.CompareFloat64(2, s)
}

// { "no": 3, "dat": "2", "ans": "2" }
func (bt *beginTask) begin3() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var s = a * b
	var p = 2 * (a + b)
	return bt.checker.CompareFloat64(2, s, p)
}

// { "no": 4, "dat": "2", "ans": "2" }
func (bt *beginTask) begin4() bool {
	var d = bt.scanner.NextFloat64()
	const PI = 3.14
	var l = PI * d
	return bt.checker.CompareFloat64(2, l)
}

// { "no": 5, "dat": "2", "ans": "3" }
func (bt *beginTask) begin5() bool {
	var a = bt.scanner.NextFloat64()
	var v = a * a * a
	var s = 6 * a * a
	return bt.checker.CompareFloat64(3, v, s)
}

// { "no": 6, "dat": "2", "ans": "3" }
func (bt *beginTask) begin6() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var v = a * b * c
	var s = 2 * (a*b + b*c + a*c)
	return bt.checker.CompareFloat64(3, v, s)
}

// { "no": 7, "dat": "2", "ans": "3" }
func (bt *beginTask) begin7() bool {
	var r = bt.scanner.NextFloat64()
	const PI = 3.14
	var l = 2 * PI * r
	var s = PI * r * r
	return bt.checker.CompareFloat64(3, l, s)
}

// { "no": 8, "dat": "2", "ans": "2" }
func (bt *beginTask) begin8() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var average = (a + b) / 2
	return bt.checker.CompareFloat64(2, average)
}

// { "no": 9, "dat": "2", "ans": "2" }
func (bt *beginTask) begin9() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var gmean = math.Sqrt(a * b)
	return bt.checker.CompareFloat64(2, gmean)
}

// { "no": 10, "dat": "2", "ans": "2" }
func (bt *beginTask) begin10() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	a *= a
	b *= b
	var sum = a + b
	var dif = a - b
	var mul = a * b
	var div = a / b
	return bt.checker.CompareFloat64(2, sum, dif, mul, div)
}

// { "no": 11, "dat": "2", "ans": "2" }
func (bt *beginTask) begin11() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	a = math.Abs(a)
	b = math.Abs(b)
	var sum = a + b
	var dif = a - b
	var mul = a * b
	var div = a / b
	return bt.checker.CompareFloat64(2, sum, dif, mul, div)
}

// { "no": 12, "dat": "2", "ans": "2" }
func (bt *beginTask) begin12() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = math.Sqrt(a*a + b*b)
	var p = a + b + c
	return bt.checker.CompareFloat64(2, c, p)
}

// { "no": 13, "dat": "2", "ans": "2" }
func (bt *beginTask) begin13() bool {
	var r1 = bt.scanner.NextFloat64()
	var r2 = bt.scanner.NextFloat64()
	const PI = 3.14
	var s1 = PI * r1 * r1
	var s2 = PI * r2 * r2
	var s3 = s1 - s2
	return bt.checker.CompareFloat64(2, s1, s2, s3)
}

// { "no": 14, "dat": "2", "ans": "2" }
func (bt *beginTask) begin14() bool {
	var l = bt.scanner.NextFloat64()
	const PI = 3.14
	var r = l / (2 * PI)
	var s = PI * r * r
	return bt.checker.CompareFloat64(2, r, s)
}

// { "no": 15, "dat": "2", "ans": "2" }
func (bt *beginTask) begin15() bool {
	var s = bt.scanner.NextFloat64()
	const PI = 3.14
	var d = math.Sqrt(4 * s / PI)
	var l = PI * d
	return bt.checker.CompareFloat64(2, d, l)
}

// { "no": 16, "dat": "2", "ans": "2" }
func (bt *beginTask) begin16() bool {
	var x1 = bt.scanner.NextFloat64()
	var x2 = bt.scanner.NextFloat64()
	var distance = math.Abs(x2 - x1)
	return bt.checker.CompareFloat64(2, distance)
}

// { "no": 17, "dat": "2", "ans": "2" }
func (bt *beginTask) begin17() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var ac = math.Abs(c - a)
	var bc = math.Abs(c - b)
	var sum = ac + bc
	return bt.checker.CompareFloat64(2, ac, bc, sum)
}

// { "no": 18, "dat": "2", "ans": "2" }
func (bt *beginTask) begin18() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var product = math.Abs(c-a) * math.Abs(c-b)
	return bt.checker.CompareFloat64(2, product)
}

// { "no": 19, "dat": "2", "ans": "2" }
func (bt *beginTask) begin19() bool {
	var x1 = bt.scanner.NextFloat64()
	var y1 = bt.scanner.NextFloat64()
	var x2 = bt.scanner.NextFloat64()
	var y2 = bt.scanner.NextFloat64()
	var a = math.Abs(x2 - x1)
	var b = math.Abs(y2 - y1)
	var p = 2 * (a + b)
	var s = a * b
	return bt.checker.CompareFloat64(2, p, s)
}

// { "no": 20, "dat": "2", "ans": "2" }
func (bt *beginTask) begin20() bool {
	var x1 = bt.scanner.NextFloat64()
	var y1 = bt.scanner.NextFloat64()
	var x2 = bt.scanner.NextFloat64()
	var y2 = bt.scanner.NextFloat64()
	var distance = math.Sqrt(math.Pow(x2-x1, 2) + math.Pow(y2-y1, 2))
	return bt.checker.CompareFloat64(2, distance)
}

// { "no": 21, "dat": "2", "ans": "2" }
func (bt *beginTask) begin21() bool {
	var x1 = bt.scanner.NextFloat64()
	var y1 = bt.scanner.NextFloat64()
	var x2 = bt.scanner.NextFloat64()
	var y2 = bt.scanner.NextFloat64()
	var x3 = bt.scanner.NextFloat64()
	var y3 = bt.scanner.NextFloat64()
	var a = math.Sqrt(math.Pow(x2-x1, 2) + math.Pow(y2-y1, 2))
	var b = math.Sqrt(math.Pow(x3-x2, 2) + math.Pow(y3-y2, 2))
	var c = math.Sqrt(math.Pow(x3-x1, 2) + math.Pow(y3-y1, 2))
	var p = a + b + c
	var halfP = p / 2
	var s = math.Sqrt(halfP * (halfP - a) * (halfP - b) * (halfP - c))
	return bt.checker.CompareFloat64(2, p, s)
}

// { "no": 22, "dat": "2", "ans": "2" }
func (bt *beginTask) begin22() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	a += b
	b = a - b
	a -= b
	return bt.checker.CompareFloat64(2, a, b)
}

// { "no": 23, "dat": "2", "ans": "2" }
func (bt *beginTask) begin23() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var d = a
	a = c
	c = b
	b = d
	return bt.checker.CompareFloat64(2, a, b, c)
}

// { "no": 24, "dat": "2", "ans": "2" }
func (bt *beginTask) begin24() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var d = a
	a = b
	b = c
	c = d
	return bt.checker.CompareFloat64(2, a, b, c)
}

// { "no": 25, "dat": "2", "ans": "2" }
func (bt *beginTask) begin25() bool {
	var x = bt.scanner.NextFloat64()
	var y = 3*math.Pow(x, 6) - 6*x*x - 7
	return bt.checker.CompareFloat64(2, y)
}

// { "no": 26, "dat": "2", "ans": "2" }
func (bt *beginTask) begin26() bool {
	var x = bt.scanner.NextFloat64()
	var thregree = math.Pow(x-3, 3)
	var y = 4*thregree*thregree - 7*thregree + 2
	return bt.checker.CompareFloat64(2, y)
}

// { "no": 27, "dat": "2", "ans": "2" }
func (bt *beginTask) begin27() bool {
	var a = bt.scanner.NextFloat64()
	var b = a * a
	if !bt.checker.CompareFloat64(2, b) {
		return false
	}
	b *= b
	if !bt.checker.CompareFloat64(2, b) {
		return false
	}
	b *= b
	return bt.checker.CompareFloat64(2, b)
}

// { "no": 28, "dat": "2", "ans": "2" }
func (bt *beginTask) begin28() bool {
	var a = bt.scanner.NextFloat64()
	var b = a * a // 2
	if !bt.checker.CompareFloat64(2, b) {
		return false
	}
	var c = b * a // 3
	if !bt.checker.CompareFloat64(2, c) {
		return false
	}
	b *= c // 5
	if !bt.checker.CompareFloat64(2, b) {
		return false
	}
	c = b * b // 10
	if !bt.checker.CompareFloat64(2, c) {
		return false
	}
	b *= c // 15
	return bt.checker.CompareFloat64(2, b)
}

// { "no": 29, "dat": "2", "ans": "2" }
func (bt *beginTask) begin29() bool {
	var deg = bt.scanner.NextFloat64()
	const PI = 3.14
	var rad = deg * PI / 180
	return bt.checker.CompareFloat64(2, rad)
}

// { "no": 30, "dat": "2", "ans": "2" }
func (bt *beginTask) begin30() bool {
	var rad = bt.scanner.NextFloat64()
	const PI = 3.14
	var deg = rad * 180 / PI
	return bt.checker.CompareFloat64(2, deg)
}

// { "no": 31, "dat": "2", "ans": "2" }
func (bt *beginTask) begin31() bool {
	var fahrenheitTemp = bt.scanner.NextFloat64()
	var centigradeTemp = (fahrenheitTemp - 32) * 5 / 9
	return bt.checker.CompareFloat64(2, centigradeTemp)
}

// { "no": 32, "dat": "2", "ans": "2" }
func (bt *beginTask) begin32() bool {
	var centigradeTemp = bt.scanner.NextFloat64()
	var fahrenheitTemp = centigradeTemp/5*9 + 32
	return bt.checker.CompareFloat64(2, fahrenheitTemp)
}

// { "no": 33, "dat": "3,2,3", "ans": "2" }
func (bt *beginTask) begin33() bool {
	var x = bt.scanner.NextFloat64()
	var a = bt.scanner.NextFloat64()
	var y = bt.scanner.NextFloat64()
	var oneKg = a / x
	var yKg = oneKg * y
	return bt.checker.CompareFloat64(2, oneKg, yKg)
}

// { "no": 34, "dat": "3,2,3,2", "ans": "2" }
func (bt *beginTask) begin34() bool {
	var x = bt.scanner.NextFloat64()
	var a = bt.scanner.NextFloat64()
	var y = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var chocoPrice = a / x
	var candyPrice = b / y
	var expensiver = chocoPrice / candyPrice
	return bt.checker.CompareFloat64(2, chocoPrice, candyPrice, expensiver)
}

// { "no": 35, "dat": "2", "ans": "2" }
func (bt *beginTask) begin35() bool {
	var v = bt.scanner.NextFloat64()
	var u = bt.scanner.NextFloat64()
	var t1 = bt.scanner.NextFloat64()
	var t2 = bt.scanner.NextFloat64()
	var s = v*t1 + (v-u)*t2
	return bt.checker.CompareFloat64(2, s)
}

// { "no": 36, "dat": "2", "ans": "2" }
func (bt *beginTask) begin36() bool {
	var v1 = bt.scanner.NextFloat64()
	var v2 = bt.scanner.NextFloat64()
	var s = bt.scanner.NextFloat64()
	var t = bt.scanner.NextFloat64()
	s += t * (v1 + v2)
	return bt.checker.CompareFloat64(2, s)
}

// { "no": 37, "dat": "2", "ans": "2" }
func (bt *beginTask) begin37() bool {
	var v1 = bt.scanner.NextFloat64()
	var v2 = bt.scanner.NextFloat64()
	var s = bt.scanner.NextFloat64()
	var t = bt.scanner.NextFloat64()
	s = math.Abs(s - t*(v1+v2))
	return bt.checker.CompareFloat64(2, s)
}

// { "no": 38, "dat": "2", "ans": "2" }
func (bt *beginTask) begin38() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var x = -b / a
	return bt.checker.CompareFloat64(2, x)
}

// { "no": 39, "dat": "2", "ans": "2" }
func (bt *beginTask) begin39() bool {
	var a = bt.scanner.NextFloat64()
	var b = bt.scanner.NextFloat64()
	var c = bt.scanner.NextFloat64()
	var d = b*b - 4*a*c
	var x1 = (-b - math.Sqrt(d)) / (2 * a)
	var x2 = (-b + math.Sqrt(d)) / (2 * a)
	return bt.checker.CompareFloat64(2, x1, x2)
}

// { "no": 40, "dat": "2", "ans": "2" }
func (bt *beginTask) begin40() bool {
	var a1 = bt.scanner.NextFloat64()
	var b1 = bt.scanner.NextFloat64()
	var c1 = bt.scanner.NextFloat64()
	var a2 = bt.scanner.NextFloat64()
	var b2 = bt.scanner.NextFloat64()
	var c2 = bt.scanner.NextFloat64()
	var d = a1*b2 - a2*b1
	var x = (c1*b2 - c2*b1) / d
	var y = (a1*c2 - a2*c1) / d
	return bt.checker.CompareFloat64(2, x, y)
}
